website/docs/basics/helpers-javascript.md

```diff
@@ -0,0 +1,311 @@
+---
+id: helpers-javascript
+title: JavaScript
+sidebar_label: JavaScript
+---
+
+[![docs-source](https://img.shields.io/badge/source-eigthshift--frontend--libs-yellow?style=for-the-badge&logo=javascript&labelColor=2a2a2a)](https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/blocks/)
+
+## checkAttr
+
+Check if attributes exist in the attributes list and add default value if not.
+
+* @param string $key Key to check.
+* @param array  $attributes Array of attributes.
+* @param array  $manifest Array of default attributes from manifest.json.
+
+**Usage:**
+
+```js
+  import { checkAttr } from '@eightshift/frontend-libs/scripts/helpers';
+
+  checkAttr('buttonUse', attributes, manifest);
+```
+
+## cookies
+
+It is used to set and get cookie values.
+
+**Usage:**
+
+```js
+  import { cookies } from '@eightshift/frontend-libs/scripts/helpers';
+
+  setCookie('gdpr', '2', cookies.setOneDay(), '/');
+
+  getCookie('gdpr');
+```
+
+## debounce
+
+Debounces the provided function.
+
+* @param {function} func Provided function to apply debounce.
+* @param {int} wait Wait time for debounce.
+
+**Usage:**
+
+```js
+  import { debounce } from '@eightshift/frontend-libs/scripts/helpers';
+
+  debounce(() => {
+    // callback function.
+  }, 250);
+```
+
+## devices
+
+Check if your browser's navigator is specific device.
+
+**Usage:**
+
+```js
+  import { device } from '@eightshift/frontend-libs/scripts/helpers';
+
+  device.iPhone();
+```
+
+## dynamicImport
+
+To get (require) all the files using the `require.context` method. It will find all files recursively in the folder using a regex. The following example will require all assets/index.js files inside the custom folder, so there is no need to manually add the files to the build.
+
+ * @param {object} paths All require.context patch to iterate.
+
+**Usage:**
+
+```js
+  import { dynamicImport } from '@eightshift/frontend-libs/scripts/helpers';
+
+  dynamicImport(require.context('./../../custom', true, /assets\/index.js$/));
+```
+
+## escape-string
+
+Takes provided string and removes special characters. Characters that will be removed: `([;&,.+*~':"!^#$%@[\]()=>|]`.
+
+* @param string $key Key to check.
+
+**Usage:**
+
+```js
+  import { escapeString } from '@eightshift/frontend-libs/scripts/helpers';
+
+  escapeString.escapeString('Special string');
+```
+
+## responsiveSelectors
+
+Create responsive selectors used for responsive attributes.
+
+* @param array   items        Array of breakpoints.
+* @param string  selector     Selector for this breakpoint.
+* @param string  parent       Parent block selector.
+* @param boolean use_modifier If false, you can use this selector for visibility.
+
+**Usage:**
+
+```js
+  import { responsiveSelectors } from '@eightshift/frontend-libs/scripts/helpers';
+
+  responsiveSelectors($attributes['width'], 'width', $blockClass);
+```
+
+**Output:**
+
+```js
+  .block-column__width-large--4
+```
+
+## selectorB
```

Comment:
couldn't this be named `selectorBEM` or `BEMselector`? `selectorB` sounds a bit like `zyklon B` 😬 And it's not clear from the name what this is.